### ISSUE TITLE
How to define loss function 

### ISSUE THREAD

**Issue by Ariel-JUAN** [iro-cp/FCRN-DepthPrediction#34]
I just use the simple L2 loss to train the model.  But the loss value can not converge.
I use tensorflow to finish it.
Here is my code:

   ` with tf.name_scope('loss'):

    labels_mask = tf.to_float(input_labels>0)

    output = net.get_output()

    loss = tf.reduce_sum(tf.square((output-input_labels)*labels_mask))/tf.reduce_sum(labels_mask)

    tf.summary.image('output', output, max_outputs=3)

    tf.summary.scalar('loss', loss)`

Is it correct? If there is something wrong, please let me know. Thanks.

**Comment by JackHenry1992** [comment]
Maybe `(output-input_labels)*labels_mask` can not get correct result (such as cannot ignore invaild depth).

**Comment by Ariel-JUAN** [comment]
@JackHenry1992  loss = tf.reduce_sum(tf.square(output-input_labels))/tf.reduce_sum(labels_mask)
Is it correct? I think it is necessary to divide by labels_mask.
